### PR TITLE
Issue417 movers refactor

### LINKED ISSUE #417: refactor .Movers to use blocks rather than multiple inheritance
This is to see how to refactor the `.Movers` models to
- avoid multiple inheritance of partial classes
- conform to the coding convention to not mix textual and graphical equations in the same class

### PR DESCRIPTION
**Pull request by mwetter**
This will close #417.

I refactored the fans so that the main equations are from blocks that are instantiated as opposed to having multiple inheritances to get the functionalities. I think the model became much easier to understand. Everything at the top-level is now graphical.

There are slight changes to the unit tests, which I updated in this pull request. In some cases, the new implementation has fewer nonlinear equations. I think this is due to heuristics of the tearing algorithm. I also tested the new implementation for the `Buildings` library, which also works fine.

I also removed the parameter `dynamicBalance` for the mover models due to https://github.com/iea-annex60/modelica-annex60/issues/411 

@Mathadon : Please review and let me know any changes, or merge to the master.

This change is not-backwards compatible. For Buildings, I use the following conversion script:
```
// Remove dynamicBalance from Movers
convertModifiers("Buildings.Fluid.Movers.SpeedControlled_y",
                 {"dynamicBalance=false"},
                 {"energyDynamics=Modelica.Fluid.Types.Dynamics.SteadyState"},
                 true);
convertModifiers("Buildings.Fluid.Movers.SpeedControlled_y",
                 {"dynamicBalance=true"},
                 {"energyDynamics=Modelica.Fluid.Types.Dynamics.FixedInitial"},
                 true);

convertModifiers("Buildings.Fluid.Movers.SpeedControlled_Nrpm",
                 {"dynamicBalance=false"},
                 {"energyDynamics=Modelica.Fluid.Types.Dynamics.SteadyState"},
                 true);
convertModifiers("Buildings.Fluid.Movers.SpeedControlled_Nrpm",
                 {"dynamicBalance=true"},
                 {"energyDynamics=Modelica.Fluid.Types.Dynamics.FixedInitial"},
                 true);

convertModifiers("Buildings.Fluid.Movers.FlowControlled_m_flow",
                 {"dynamicBalance=false"},
                 {"energyDynamics=Modelica.Fluid.Types.Dynamics.SteadyState"},
                 true);
convertModifiers("Buildings.Fluid.Movers.FlowControlled_m_flow",
                 {"dynamicBalance=true"},
                 {"energyDynamics=Modelica.Fluid.Types.Dynamics.FixedInitial"},
                 true);

convertModifiers("Buildings.Fluid.Movers.FlowControlled_dp",
                 {"dynamicBalance=false"},
                 {"energyDynamics=Modelica.Fluid.Types.Dynamics.SteadyState"},
                 true);
convertModifiers("Buildings.Fluid.Movers.FlowControlled_dp",
                 {"dynamicBalance=true"},
                 {"energyDynamics=Modelica.Fluid.Types.Dynamics.FixedInitial"},
                 true);

// Remove r_N.start which is no longer a public variable in .Movers
convertModifiers({"Buildings.Fluid.Movers.SpeedControlled_y",
                  "Buildings.Fluid.Movers.SpeedControlled_Nrpm",
                  "Buildings.Fluid.Movers.FlowControlled_m_flow",
                  "Buildings.Fluid.Movers.FlowControlled_dp"},
                  {"r_N.start"},
                  fill("", 0),
                  true);
convertModifiers({"Buildings.Fluid.Movers.BaseClasses.PartialFlowMachine"},
                  {"addPowerToMedium"},
                  fill("", 0),
                  true);


// We remove the parameter setting below because this parameter is
// already assigned in the performance record per.
convertModifiers("Buildings.Fluid.Movers.FlowControlled_dp",
               {"use_powerCharacteristic"},
               fill("",0));
convertModifiers("Buildings.Fluid.Movers.FlowControlled_m_flow",
               {"use_powerCharacteristic"},
               fill("",0));
convertModifiers("Buildings.Fluid.Movers.SpeedControlled_Nrpm",
               {"use_powerCharacteristic"},
               fill("",0));
convertModifiers("Buildings.Fluid.Movers.SpeedControlled_y",
               {"use_powerCharacteristic"},
               fill("",0));

convertModifiers("Buildings.Fluid.Movers.FlowControlled_dp",
               {"motorCooledByFluid"},
               fill("",0));
convertModifiers("Buildings.Fluid.Movers.FlowControlled_m_flow",
               {"motorCooledByFluid"},
               fill("",0));
convertModifiers("Buildings.Fluid.Movers.SpeedControlled_Nrpm",
               {"motorCooledByFluid"},
               fill("",0));
convertModifiers("Buildings.Fluid.Movers.SpeedControlled_y",
               {"motorCooledByFluid"},
               fill("",0));

// Update data records for movers
convertModifiers("Buildings.Fluid.Movers.Data.SpeedControlled_Nrpm",
                 {"N_nominal=1500"}, {"speed_rpm_nominal=%N_nominal%"});
convertClass("Buildings.Fluid.Movers.Data.FlowControlled",
             "Buildings.Fluid.Movers.Data.Generic");
convertClass("Buildings.Fluid.Movers.Data.SpeedControlled_Nrpm",
             "Buildings.Fluid.Movers.Data.Generic");
convertClass("Buildings.Fluid.Movers.Data.SpeedControlled_y",
             "Buildings.Fluid.Movers.Data.Generic");

```

### CODE REVIEW

File: Annex60/Fluid/Movers/BaseClasses/EfficiencyInterface.mo [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+              {7,21}},
+          lineColor={0,0,0},
+          fillColor={100,100,100},
+          fillPattern=FillPattern.Solid,
+          smooth=Smooth.Bezier,
+          origin={15,7},
+          rotation=90)}),
+    Documentation(info="<html>
+<p>
+Blcok that computes the mover efficiency and power consumption.
```
Comment (Mathadon) [inline Annex60/Fluid/Movers/BaseClasses/EfficiencyInterface.mo]: typo

**Comment by Mathadon** [comment]
I verified your changes. Todo: check for further simplifications.

**Comment by mwetter** [comment]
@mlauster Can you please have a last review at this mover refactoring that @Mathadon and I did and merge to the master if all is fine. All unit tests work in pedantic mode.
Note the non-backward compatible changes as described in the above notes.

**Comment by mwetter** [comment]
@mlauster Please review but do not merge yet as OpenModelica seems to have problems with flattening, which I am looking into.

**Comment by mwetter** [comment]
@mlauster : I fixed the translation problems of OpenModelica. Now all examples and validations in `Annex60.Movers` are working with OpenModelica. If you are OK with your review, the code is from my point ready to be merged to the `master`.

**Comment by mwetter** [comment]
I also added the `final alpha=0` to the prescribed heat input of the fan in cca2faf
This will close https://github.com/iea-annex60/modelica-annex60/issues/399

Commit fbb74d7 updates the reference results as `final alpha=0` leads to smaller systems of equations.

**Pull request merged**